Source/Mock.cs

```diff
@@ -834,42 +834,22 @@ private static void SetupAllProperties(Mock mock, Stack<Type> mockedTypesStack)
 					   ProxyFactory.Instance.IsMethodVisible(p.GetGetMethod(), out _))
 				.Distinct();
 
-			var setupPropertyMethod = mock.GetType().GetMethods("SetupProperty")
-				.First(m => m.GetParameters().Length == 2);
-			var setupGetMethod = mock.GetType().GetMethods("SetupGet")
-				.First(m => m.GetParameters().Length == 1);
-
 			foreach (var property in properties)
 			{
 				var expression = GetPropertyExpression(mockType, property);
-				object initialValue = GetInitialValue(mock, mockedTypesStack, property);
+				object value = GetInitialValue(mock, mockedTypesStack, property);
 
-				var mocked = initialValue as IMocked;
+				var mocked = value as IMocked;
 				if (mocked != null)
 				{
 					SetupAllProperties(mocked.Mock, mockedTypesStack);
 				}
 
+				mock.Setups.Add(new PropertyGetterMethodCall(mock, expression, property.GetGetMethod(true), () => value));
```

Comment:
The above LINQ query only returns properties that are readable, so we don't need to check `property.CanRead`.